### ISSUE TITLE
False SAT point found by Marabou

### ISSUE THREAD

**Issue by kjulian3** [NeuralNetworkVerification/Marabou#325]
While creating benchmark properties, I created a property that spans the full input space and searches for a point where the 5th output (y4) is maximum, and Marabou returned a SAT point 
(x0=-0.033329, x1=-0.046218, x2=0.098878). However, when I evaluated this point, I found that y0 is the maximum output, not y4. Furthermore, if I shrink the search space to be a small region centered around the SAT point found by searching the full input space, Marabou returns UNSAT. So it seems that the original SAT point found by Marabou is not correct, and the result should not be SAT.

I rebuilt Marabou just now and still see the issue. I've attached the neural network file as well as three property files:
* 5.1_SAT.txt: Full input search space with y4 is maximum property set, which produces a SAT point where y4 is the maximum output
* 5.2_UNSAT.txt: Reduced search space around the SAT point, which produces UNSAT
* 5.3_SAT.txt: Constraining input variables to the SAT point without an output constraint, which produces outputs showing that y0 is the maximum output, not y4.

[MarabouIssue.zip](https://github.com/NeuralNetworkVerification/Marabou/files/4923145/MarabouIssue.zip)

**Comment by kjulian3** [comment]
After more investigation, I found that commit 53f04fb0a8b7ca28aa0edf721367f1ac9891cf03 returns a SAT point for 5.1_SAT.txt that is not accurate (runtime was 5 minutes), while the previous commit does not return SAT after ten minutes of running.

The query was run using 
```
./Marabou MarabouIssue/HCAS_rect_v6_pra0_tau60_25HU_3000.nnet MarabouIssue/5.1_SAT.txt
```

The SAT point found is:
Input assignment:
        x0 = -0.035147
        x1 = -0.047176
        x2 = 0.100550
Output:
        y0 = 0.013583
        y1 = 0.012902
        y2 = 0.013218
        y3 = 0.010654
        y4 = 0.013583

This shows that y4 greater than or equal to all other output variables. However, if you constrain the input variables to be exactly the SAT point and remove the output constraint, the output variables are:
Input assignment:
        x0 = -0.035147
        x1 = -0.047176
        x2 = 0.100550
Output:
        y0 = 0.014331
        y1 = 0.012030
        y2 = 0.013189
        y3 = 0.010514
        y4 = 0.013512
These output values significantly differ from those of the original 5.1_SAT.txt property, even though the input assignments are the same. These output values show that y0 is the maximum output, not y4, and I verified that these output values are correct using functions in the NNet repository.

**Comment by kjulian3** [comment]
The commit just before the commit referenced above also returned SAT after 11 minutes of running on my machine, so there still seems to be a problem. With the most recent version of Marabou, an illegitimate SAT point can be found in a little over a minute (on my machine).

**Comment by anwu1219** [comment]
This commit 1e09fe47537e0dc448b0eb786c333844ffc4e9d6 already seems to return a false positive.
It appears that the issue could be caused by the choices of the precision-related hyper-parameters. 
I created this branch https://github.com/anwu1219/Marabou/tree/higher-precision with higher precision parameters, which seems to find true positive.

**Comment by kjulian3** [comment]
This issue was solved by reducing tolerance of relu constraints, which was addressed in #329. So this issue is solved.
